base/precompilation.jl

```diff
@@ -43,12 +43,12 @@ function ExplicitEnv(envpath::String=Base.active_project())
 
     # Collect all direct dependencies of the project
     for key in ["deps", "weakdeps", "extras"]
-        for (name, _uuid::String) in get(Dict{String, Any}, project_d, key)::Dict{String, Any}
```

Comment:
It is a shame that this behaves like:
```julia
_uuid::String
_uuid = unpacked_val # implicit `convert`
```
instead of:
```julia
_uuid = unpacked_val::String
```
or even:
```julia
_uuid::String = unpacked_val::String
```

The original author of this code clearly thought that was how this worked, and TBH so did I until I looked at the `code_typed`

Reply:
This is all @KristofferC's new code that allowed parallel precompilation to leave Pkg. Just tagging to check intention on these converts, as I've been marked for review.

Reply:
FWIW I'm pretty confident the new annotations are correct - I just wanted to leave a note behind about the weird semantics here (it almost feels like a lowering bug that it behaves this way)